plugins/storage/volume/linstor/src/main/java/com/cloud/hypervisor/kvm/storage/LinstorStorageAdaptor.java

```diff
@@ -311,12 +313,13 @@ public boolean connectPhysicalDisk(String volumePath, KVMStoragePool pool, Map<S
             throw new CloudRuntimeException(apiEx.getBestMessage(), apiEx);
         }
 
-        try
-        {
-            allow2PrimariesIfInUse(api, rscName);
-        } catch (ApiException apiEx) {
-            s_logger.error(apiEx);
-            // do not fail here as adding allow-two-primaries property is only a problem while live migrating
+        if (isVMMigration) {
+            try {
+                allow2PrimariesIfInUse(api, rscName);
+            } catch (ApiException apiEx) {
+                s_logger.error(apiEx);
+                // do not fail here as adding allow-two-primaries property is only a problem while live migrating
+            }
```

Comment:
Could you explain a bit further how did you reproduce the issue, and how you verified that it no longer happens with this change?

Reply:
The simple check is, to see if on a cold start to see if the new debug line is in the logs.
It should only be visible on live migrated (which it was on my tests).
For the actual issue that started this, it is a bit more complicated and I couldn't fully reproduce it yet, as you need to let CloudStack start VM HA on a host, but the host shouldn't be fenced of. so basically freezing the CloudStack agent and having the other hosts say that this host is down, even tough it isn't.
Which can happen if you use a non hyperconverged setup with the current Linstor HA code, there will be a follow up PR to fix this aswell.

Reply:
Just to see if I got it right: your problem was that the agent got stuck, the other hosts said it was down and the 'HA' was triggered, so the VM was started on another host, even though it was already running in the host that got stuck?

If so, on Linstor, in this process, why would the VM be using two primaries (if this is what's happening)?

Reply:
Ok, I think I got it